### DISCUSSION TITLE
How would i make the music selector seen in sonic generations for my project

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#78] [Q&A]
How would i make the music selector seen in sonic generations for my project.

I have tried to do it my self but it didn't work

**Comment by DimensionWarped** [comment]
When I say you should be able to detail what you have tried so far, Im saying that you should at least talk about other pieces of script and/or scenes you have looked at while trying to figure it out.

**Reply by Raptorq1** [reply]
if Global.music != null:
		if music != null:
			Global.music.stream = music
			Global.music.play()
			Global.music.stream_paused = false
		else:
			Global.music.stop()
			Global.music.stream = null
			
		if Global.music !=null and Global.music_style ==1:
			if music2 !=null:
				Global.music.stream = music2
				Global.music.play()
				Global.music.stream_paused = false
			else:
				Global.music.stop()
				Global.music.stream =null
				
				if Global.music  != null and Global.extras == 1:
					if music3 != null:
						Global.music.stream = music3
						Global.music.play()

**Comment by DimensionWarped** [comment]
And where exactly does this code live? And what were the results? Have you tried probing to see what's going on with logs?

For that matter, what specifically are you trying to accomplish? My immediate guess is something like how Sonic 3 Air lets you pick between the original track or a secondary track for certain zones that had alternative music in certain releases. Is that the case?

**Reply by Raptorq1** [reply]
Its in the level script, and yes that's what I'm trying to do

**Comment by DimensionWarped** [comment]
Ok, and where are you setting your global music values? Also did you use logs to try to figure out where things are going wrong? Usually you can figure out a lot just by printing the string values of certain variables.

**Reply by Raptorq1** [reply]
I can't use logs because I'm on mobile and I'm setting the global music values in Global.gd

**Reply by Techokami** [reply]
I uh... I didn't even know Godot *had* a mobile version of the IDE... we've all been using regular desktop computers to make this, which have the best possible set of tools for things like debugging. It sounds like you're doing this on Hard Mode!

**Reply by DimensionWarped** [reply]
I don't know much about Godot 4 mobile, but I would wager that it has *some* debug tools or else they wouldn't have bothered making it. That said, it doesn't really matter... you are going to *need* to start doing this with a computer so that you have things like windows and proper multitasking (IE not phone multitasking) for the sake of your own sanity. I think Godot can run on fairly low end equipment (probably a used 200 dollar office e-waste thinkpad laptop would be more than adequate), so we're not talking anything too expensive, though I know the definition of too expensive varies a lot from person to person and from place to place.

I'm including an invite link to an adhoc discord server if you want to get in touch for a one-on-one discussion. We're talking about creating a Discord server specifically for this project, but it's not here yet, so this is the best way I can think of to get in touch. You can usually find me between 9:00AM - 12:00AM, though I do tend to be unresponsive for long periods of time while I'm working.

https://discord.gg/bXdHy66j

**Reply by Raptorq1** [reply]
Yep it does have them but when I run the project I can't see them (p.s thanks for all your help🙂)

**Reply by Raptorq1** [reply]
I found the problem its the engine it's self not allowing for the music to play
